google-ads-examples/src/main/java/com/google/ads/googleads/examples/shoppingads/AddPerformanceMaxProductListingGroupTree.java

```diff
@@ -0,0 +1,554 @@
+        for (String child : children) {
+          operations.addAll(removeDescendantsAndFilter(child));
+        }
+      }
+
+      AssetGroupListingGroupFilterOperation operation =
+          AssetGroupListingGroupFilterOperation.newBuilder().setRemove(resourceName).build();
+      operations.add(
+          MutateOperation.newBuilder().setAssetGroupListingGroupFilterOperation(operation).build());
+      return operations;
+    }
+  }
+  // [END add_performance_max_product_listing_group_tree_3]
+
+  /**
+   * A factory that creates MutateOperations wrapping AssetGroupListingGroupFilterMutateOperations
+   * for a specific customerId and assetGroupId.
+   *
+   * <p>This object is intended to be used with an array of MutateOperations to perform an atomic
+   * update to an AssetGroup.
+   */
+  private static class AssetGroupListingGroupFilterCreateOperationFactory {
+    private final long customerId;
+    private final long assetGroupId;
+    private final long rootListingGroupId;
+    private long nextId;
+
+    public AssetGroupListingGroupFilterCreateOperationFactory(
+        long customerId, long assetGroupId, long rootListingGroupId) {
+      this.customerId = customerId;
+      this.assetGroupId = assetGroupId;
+      this.rootListingGroupId = rootListingGroupId;
+      this.nextId = this.rootListingGroupId;
+    }
+
+    /**
+     * Returns a new temporary ID to be used for a resource name in a MutateOperation. See
+     * https://developers.google.com/google-ads/api/docs/mutating/best-practices#temporary_resource_names
+     * for details about temporary IDs.
+     */
+    public long getNextId() {
+      return --nextId;
+    }
+
+    // [START add_performance_max_product_listing_group_tree_4]
+    /**
+     * Creates a MutateOperation that creates a root AssetGroupListingGroupFilter for the factory's
+     * AssetGroup.
+     *
+     * <p>The root node or partition is the default, which is displayed as "All Products".
+     */
+    public MutateOperation createRoot() {
+      AssetGroupListingGroupFilter listingGroupFilter =
+          AssetGroupListingGroupFilter.newBuilder()
+              .setResourceName(
+                  ResourceNames.assetGroupListingGroupFilter(
+                      customerId, assetGroupId, rootListingGroupId))
+              .setAssetGroup(ResourceNames.assetGroup(customerId, assetGroupId))
+              // Since this is the root node, do not set the ParentListingGroupFilter. For all other
+              // nodes, this would refer to the parent listing group filter resource name.
+              // ParentListingGroupFilter = "<PARENT FILTER NAME>"
+              //
+              // Unlike AddPerformanceMaxRetailCampaign, the type for the root node here must be
+              // Subdivision because we add child partitions under it.
+              .setType(ListingGroupFilterType.SUBDIVISION)
+              // Because this is a Performance Max campaign for retail, we need to specify that this
```

Comment:
Are the "we" sentences seem stylistically inconsistent with other examples in the Java lib? WDYT of reframing like `Specifies this is in the shopping vertical, as required for a Performance Max campaign for retail` or something similar.

Basically, following the implicit "This code..." prefix pattern we use elsewhere.